### DISCUSSION TITLE
How to get the list of functions in a library .so

### DISCUSSION THREAD

**Discussion by leopeuziat** [thelfer/tfel#167] [Q&A] [answered]
I would like to use the MTest Python API to plot a material property as a function of its parameters. In a library .so there can be several functions. For exemple, in the VanadiumAlloy material given as an exemple in the MFront Gallery, there are PoissonRatio, ShearModulus, SpecificHeat... Is it possible to get the list of functions in a library .so ?
Thanks by advance,

**Comment by thelfer** [comment, marked as answer]
@leopeuziat You can use the following `python` script:

```cxx
from tfel.system import ExternalLibraryManager
elm = ExternalLibraryManager.getExternalLibraryManager()
print(elm.getEntryPoints('vanadiumalloy.so'))
```

This works with any library generated by MFront.

The output is:

```
['VanadiumAlloy_YoungModulus_SRMA', 'VanadiumAlloy_SpecificHeat_SRMA', 'VanadiumAlloy_PoissonRatio_SRMA', 'VanadiumAlloy_ShearModulus_SRMA', 'VanadiumAlloy_ThermalExpansion_SRMA', 'VanadiumAlloy_ThermalConductivity_SRMA']
```
